src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -287,6 +287,18 @@ internal class MemberSignatureComparer : IEqualityComparer<Symbol>
             considerRefKindDifferences: true,
             typeComparison: TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds | TypeCompareKind.IgnoreNullableModifiersForReferenceTypes | TypeCompareKind.IgnoreDynamicAndTupleNames);
 
+        /// <summary>
+        /// Compare signatures of methods from a method group.
+        /// </summary>
+        internal static readonly MemberSignatureComparer MethodGroupSignatureComparer = new MemberSignatureComparer(
+            considerName: false,
+            considerExplicitlyImplementedInterfaces: false,
+            considerReturnType: true,
+            considerTypeConstraints: false,
+            considerRefKindDifferences: true,
+            considerCallingConvention: false,
```

Comment:
>considerCallingConvention: false [](start = 12, length = 32)

I'm not sure about this one. Consider something like:

```cs
var s = new S();
var x = &s.M;

struct S {}

static class C1
{
    [UnmanagedCallersOnly(typeof(CallConvCdecl))]
    public static void M(this S s) {}
}

static class C2
{
    [UnmanagedCallersOnly(typeof(CallConvStdcall))]
    public static void M(this S s) {}
}
```

I don't think I'd expect the compiler to be able to infer a common type for this case, but I think we would given this comparison option.

Reply:
(Or the equivalent without a function pointer, `var x = s.M;`

---
In reply to: [618595428](https://github.com/dotnet/roslyn/pull/52448#discussion_r618595428) [](ancestors = 618595428)

Reply:
I've added a couple of tests although it's not clear to me that there's an issue since `[UnmanagedCallersOnly]` methods cannot be converted to delegates.